edn/src/edn.rustpeg

```diff
@@ -44,13 +44,31 @@ pub boolean -> Value =
     "false" { Value::Boolean(false) }
 
 digit = [0-9]
+alphanumeric = [0-9a-zA-Z]
+octaldigit = [0-7]
+hex = [0-9a-fA-F]
 sign = "-" / "+"
 
 pub bigint -> Value =
     b:$( sign? digit+ ) "N" {
         Value::BigInteger(b.parse::<BigInt>().unwrap())
     }
 
+pub octalinteger -> Value =
+    "0" i:$( octaldigit+ ) {
+        Value::Integer(i64::from_str_radix(i, 8).unwrap())
+    }
+
+pub hexinteger -> Value =
+    "0x" i:$( hex+ ) {
+        Value::Integer(i64::from_str_radix(i, 16).unwrap())
+    }
+
+pub basedinteger -> Value =
+    b:$(digit digit?) "r" i:$( alphanumeric+ ) {
```

Comment:
One option is to define the acceptable bases more accurately:

```
[2..9] / [12][0-9] / 3[0..6]
```

Another is to figure out how rust-peg expects rules to not apply. I believe that if you do this:

```
  b:$(…) … $(alphanumeric+) {?
      let base = b.parse::<u32>()?;
      let num = i64::from_str_radix(i, base)?
      Value::Integer(num)
  }
```

it'll do what you want.